altair/examples/co2_concentration.py

```diff
@@ -0,0 +1,66 @@
+"""
+Iowa's renewable energy boom
+----------------------------
+This example is a fully developed line chart that uses a window transformation. It was inspired by `Gregor Aisch's work at datawrapper <https://www.datawrapper.de/_/OHgEm/>_`.
+"""
+# category: case studies
+import altair as alt
+from vega_datasets import data
+
+source = data.co2_concentration()
+
+base = alt.Chart(
+    source,
+    title="Carbon Dioxide in the Atmosphere"
+).transform_calculate(
+    year="year(datum.Date)"
+).transform_calculate(
+    decade="floor(datum.year / 10)"
+).transform_calculate(
+    scaled_date="(datum.year % 10) + (month(datum.Date)/12)"
+).transform_window(
+    window=[
+        {'op': 'first_value', 'as': 'first_date', 'field': 'scaled_date'},
+        {'op': 'last_value', 'as': 'last_date', 'field': 'scaled_date'}
+    ],
+    sort=[
+        {"field": "scaled_date", "order": "ascending"},
+        {"field": "scaled_date", "order": "ascending"}
+    ],
+    groupby=["decade"],
+    frame=[None, None]
+).transform_calculate(
+  end="datum.first_date === datum.scaled_date ? 'first' : datum.last_date === datum.scaled_date ? 'last' : null"
+).encode(
+    x=alt.X(
+        "scaled_date:Q",
+        axis=alt.Axis(title="Year into Decade", tickCount=11)
+    ),
+    y=alt.Y(
+        "CO2:Q",
+        title="CO2 concentration in ppm",
+        scale=alt.Scale(zero=False)
+    )
+)
+
+line = base.mark_line().encode(
+    color=alt.Color(
+        "decade:O",
+        scale=alt.Scale(scheme="magma"),
+        legend=None
+    )
+)
+
+start_year = base.transform_filter(
+  {"field": "end", "equal": "first"}
+).mark_text(baseline="top").encode(text="year:N")
+
+end_year = base.transform_filter(
+  {"field": "end", "equal": "last"}
+).mark_text(baseline="bottom").encode(text="year:N")
```

Comment:
Instead of passing a dict to ``transform_filter`` we should use expressions: ``{"field": "end", "equal": "first"}`` can be written ``alt.datum.end == 'first'``... similar with `'last'`